src/lib/Microsoft.Health.Extensions.Fhir.R4/FhirClientFactory.cs

```diff
@@ -105,6 +114,13 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                 var accessToken = await TokenCredential.GetTokenAsync(requestContext, CancellationToken.None);
                 request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", accessToken.Token);
                 var response = await base.SendAsync(request, cancellationToken);
+
+                if (Logger != null && !response.IsSuccessStatusCode)
+                {
+                    var statusDescription = response.ReasonPhrase.Replace(" ", string.Empty);
+                    Logger.LogMetric(FhirClientMetrics.HandledException($"FhirServerError{statusDescription}", ConnectorOperation.FHIRConversion), 1);

```

Comment:
Do we know how many different status descriptions we can expect?

Reply:
I had not tried to generate every possible http response, but wanted to capture any non-success error that could be returned. The main errors expect are "Forbidden", "Unauthorized", "Too Many Requests", and "Internal Server Error".

Reply:
Thanks,  We may want to have a special classification for Forbidden/Unauthorized so we can really make them stand out but we can do that later if necessary.